core-primitives/node-api-extensions/src/metadata.rs

```diff
@@ -0,0 +1,190 @@
+		self.storage_value_key(SYSTEM, "Account")
+	}
+	pub fn system_account_storage_map_key(&self, index: u64) -> Result<StorageKey> {
+		self.storage_map_key(SYSTEM, "Account", &index)
+	}
+
+	/// Get the storage keys corresponding to a storage over the metadata
+	/// Storage value key
+	pub fn storage_value_key(
+		&self,
+		storage_prefix: &'static str,
+		storage_key_name: &'static str,
+	) -> Result<StorageKey> {
+		match &self.node_metadata {
+			None => return Err(Error::MetadataNotSet),
+			Some(m) => m
+				.storage_value_key(storage_prefix, storage_key_name)
+				.map_err(|e| Error::NodeMetadata(e)),
+		}
+	}
+	/// Storage map key
+	pub fn storage_map_key<K: Encode>(
+		&self,
+		storage_prefix: &'static str,
+		storage_key_name: &'static str,
+		map_key: K,
+	) -> Result<StorageKey> {
+		match &self.node_metadata {
+			None => return Err(Error::MetadataNotSet),
+			Some(m) => m
+				.storage_map_key::<K>(storage_prefix, storage_key_name, map_key)
+				.map_err(|e| Error::NodeMetadata(e)),
+		}
+	}
+	/// Storage double map key
+	pub fn storage_double_map_key<K: Encode, Q: Encode>(
+		&self,
+		storage_prefix: &'static str,
+		storage_key_name: &'static str,
+		first: K,
+		second: Q,
+	) -> Result<StorageKey> {
+		match &self.node_metadata {
+			None => return Err(Error::MetadataNotSet),
+			Some(m) => m
+				.storage_double_map_key(storage_prefix, storage_key_name, first, second)
+				.map_err(|e| Error::NodeMetadata(e)),
+		}
+	}
+}
+pub trait NodeRuntimeVersion {
+	/// Return the substrate chain runtime version
+	fn get_runtime_version(&self) -> u32;
+
+	/// Return the substrate chain runtime transaction version
+	fn get_runtime_transaction_version(&self) -> u32;
+}
+
+/// Get the array [pallet index, call index] corresponding to a pallet's call over the metadata.
+pub trait NodeCallsIndexes {
+	/// Teerex Calls
+	fn register_enclave_call_indexes(&self) -> Result<[u8; 2]>;
+
+	fn unregister_enclave_call_indexes(&self) -> Result<[u8; 2]>;
+
+	fn call_worker_call_indexes(&self) -> Result<[u8; 2]>;
+
+	fn confirm_processed_parentchain_block_call_indexes(&self) -> Result<[u8; 2]>;
+
+	fn shield_funds_call_indexes(&self) -> Result<[u8; 2]>;
+
+	fn unshield_funds_call_indexes(&self) -> Result<[u8; 2]>;
+
+	/// Sidechain Calls
+	fn confirm_proposed_sidechain_block_indexes(&self) -> Result<[u8; 2]>;
+
+	/// General
+	fn call_indexes(&self, pallet_name: &'static str, call_name: &'static str) -> Result<[u8; 2]>;
+}
+
+impl NodeRuntimeVersion for NodeMetadata {
+	fn get_runtime_version(&self) -> u32 {
+		self.runtime_spec_version
+	}
+
+	fn get_runtime_transaction_version(&self) -> u32 {
+		self.runtime_transaction_version
+	}
+}
+
+impl NodeCallsIndexes for NodeMetadata {
+	fn register_enclave_call_indexes(&self) -> Result<[u8; 2]> {
+		self.call_indexes(TEEREX, "register_enclave")
+	}
+	fn unregister_enclave_call_indexes(&self) -> Result<[u8; 2]> {
+		self.call_indexes(TEEREX, "unregister_enclave")
+	}
```

Comment:
Please add an empty line between function definitions (in some instances you have one, but mostly not)